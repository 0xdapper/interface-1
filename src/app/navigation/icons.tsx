import { useTheme } from '@shopify/restyle'
import React from 'react'
import SearchIcon from 'src/assets/icons/search.svg'
import WalletIcon from 'src/assets/icons/wallet.svg'
import { Theme } from 'src/styles/theme'

interface NavIconProps {
  focused: boolean
}

export function WalletTabIcon({ focused }: NavIconProps) {
  const theme = useTheme<Theme>()

  return (
    <WalletIcon fill={theme.colors.textColor} height={24} opacity={focused ? 1 : 0.6} width={24} />
  )
}

export function ExploreTabIcon({ focused }: NavIconProps) {
  const theme = useTheme<Theme>()

  return <SearchIcon opacity={focused ? 1 : 0.6} stroke={theme.colors.textColor} strokeWidth={2} />
}
