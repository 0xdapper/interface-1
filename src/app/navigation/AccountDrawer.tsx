import { DrawerContentComponentProps } from '@react-navigation/drawer'
import { default as React, useMemo, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { ListRenderItemInfo } from 'react-native'
import { useAppDispatch, useAppTheme } from 'src/app/hooks'
import PlusSquareIcon from 'src/assets/icons/plus-square.svg'
import SettingsIcon from 'src/assets/icons/settings.svg'
import { AccountCardItem } from 'src/components/accounts/AccountCardItem'
import { RemoveAccountModal } from 'src/components/accounts/RemoveAccountModal'
import { Button } from 'src/components/buttons/Button'
import { AnimatedFlex, Box, Flex } from 'src/components/layout'
import { AnimatedFlatList } from 'src/components/layout/AnimatedFlatList'
import { Screen } from 'src/components/layout/Screen'
import { ActionSheetModal, MenuItemProp } from 'src/components/modals/ActionSheetModal'
import { BottomSheetModal } from 'src/components/modals/BottomSheetModal'
import { Text } from 'src/components/Text'
import { WalletQRCode } from 'src/components/WalletConnect/ScanSheet/WalletQRCode'
import { importAccountActions } from 'src/features/import/importAccountSaga'
import { ElementName, ModalName } from 'src/features/telemetry/constants'
import { Account, AccountType } from 'src/features/wallet/accounts/types'
import { EditAccountAction, editAccountActions } from 'src/features/wallet/editAccountSaga'
import { useAccounts, useActiveAccount } from 'src/features/wallet/hooks'
import { activateAccount } from 'src/features/wallet/walletSlice'
import { Screens } from 'src/screens/Screens'
import { setClipboard } from 'src/utils/clipboard'

const key = (account: Account) => account.address

export function AccountDrawer({ navigation }: DrawerContentComponentProps) {
  const { t } = useTranslation()
  const theme = useAppTheme()

  const activeAccount = useActiveAccount()
  const addressToAccount = useAccounts()
  const dispatch = useAppDispatch()

  const [qrCodeAddress, setQRCodeAddress] = useState(activeAccount?.address)
  const [showQRModal, setShowQRModal] = useState(false)
  const [showEditAccountModal, setShowEditAccountModal] = useState(false)
  const [pendingEditAddress, setPendingEditAddress] = useState<Address | null>(null)

  const accountsData = useMemo(() => {
    const accounts = Object.values(addressToAccount)
    const _signerAccounts = accounts.filter((a) => a.type !== AccountType.Readonly)
    const _readOnlyAccounts = accounts.filter((a) => a.type === AccountType.Readonly)
    return [..._signerAccounts, ..._readOnlyAccounts]
  }, [addressToAccount])

  const onPressEdit = (address: Address) => {
    setShowEditAccountModal(true)
    setPendingEditAddress(address)
  }
  const onPressEditCancel = () => {
    setShowEditAccountModal(false)
    setPendingEditAddress(null)
  }

  const [pendingRemoveAddress, setPendingRemoveAddress] = useState<Address | null>(null)
  const onPressRemoveCancel = () => {
    setPendingRemoveAddress(null)
  }
  const onPressRemoveConfirm = () => {
    if (!pendingRemoveAddress) return
    dispatch(
      editAccountActions.trigger({ type: EditAccountAction.Remove, address: pendingRemoveAddress })
    )
    setPendingRemoveAddress(null)
    onPressEditCancel() // Dismiss bottom sheet
  }

  const onPressAccount = (address: Address) => {
    navigation.closeDrawer()
    dispatch(activateAccount(address))
  }

  const onPressQRCode = (address: Address) => {
    setQRCodeAddress(address)
    setShowQRModal(true)
  }

  const onCloseQrCode = () => setShowQRModal(false)

  const onPressNewAccount = () => {
    // First reset to clear saga state that's left over from dev account import
    // TODO remove when use of dev account is removed
    dispatch(importAccountActions.reset())
    navigation.navigate(Screens.ImportAccount)
  }

  const onPressSettings = () => {
    navigation.closeDrawer()
    navigation.navigate(Screens.SettingsStack, { screen: Screens.Settings })
  }

  const renderItem = ({ item }: ListRenderItemInfo<Account>) => {
    return (
      <AccountCardItem
        account={item}
        isActive={!!activeAccount && activeAccount.address === item.address}
        isViewOnly={item.type === AccountType.Readonly}
        onPress={onPressAccount}
        onPressEdit={onPressEdit}
        onPressQRCode={onPressQRCode}
      />
    )
  }

  const header = (
    <AnimatedFlex bg="mainBackground" borderBottomColor="backgroundOutline" px="lg" py="sm">
      <Text color="textPrimary" variant="headlineSmall">
        {t('Your wallets')}
      </Text>
    </AnimatedFlex>
  )

  const editAccountOptions = useMemo<MenuItemProp[]>(() => {
    const onPressWalletSettings = () => {
      setShowEditAccountModal(false)
      navigation.closeDrawer()
      if (!pendingEditAddress) return
      navigation.navigate(Screens.SettingsStack, {
        screen: Screens.SettingsWallet,
        params: { address: pendingEditAddress },
      })
    }

    const onPressCopyAddress = () => {
      if (!pendingEditAddress) return
      setClipboard(pendingEditAddress)
      setShowEditAccountModal(false)
    }

    const onPressRemove = () => {
      if (!pendingEditAddress) return
      setShowEditAccountModal(false)

      // For view-only wallets, we don't show the remove wallet modal and instead remove immediately
      if (addressToAccount[pendingEditAddress].type === AccountType.Readonly) {
        dispatch(
          editAccountActions.trigger({
            type: EditAccountAction.Remove,
            address: pendingEditAddress,
          })
        )
      } else {
        setPendingRemoveAddress(pendingEditAddress)
      }
    }

    return [
      {
        key: ElementName.WalletSettings,
        onPress: onPressWalletSettings,
        render: () => (
          <Box alignItems="center" p="md">
            <Text variant="body">{t('Wallet settings')}</Text>
          </Box>
        ),
      },
      {
        key: ElementName.Copy,
        onPress: onPressCopyAddress,
        render: () => (
          <Box alignItems="center" p="md">
            <Text variant="body">{t('Copy address')}</Text>
          </Box>
        ),
      },
      {
        key: ElementName.Remove,
        onPress: onPressRemove,
        render: () => (
          <Box alignItems="center" p="md">
            <Text color="accentFailure" variant="body">
              {t('Remove account')}
            </Text>
          </Box>
        ),
      },
    ]
  }, [navigation, pendingEditAddress, t, addressToAccount, dispatch])

  return (
    <Screen bg="backgroundBackdrop" width="100%">
      <AnimatedFlatList
        ListHeaderComponent={header}
        data={accountsData}
        keyExtractor={key}
        renderItem={renderItem}
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        stickyHeaderIndices={[0]}
      />
      <Flex>
        <Box bg="backgroundOutline" height={0.5} mb="sm" />
        <Flex gap="xl" pb="xl" px="lg">
          <Button
            name={ElementName.ImportAccount}
            testID={ElementName.ImportAccount}
            onPress={onPressNewAccount}>
            <Flex row alignItems="center" gap="sm">
              <PlusSquareIcon color={theme.colors.textSecondary} height={24} width={24} />
              <Text color="textSecondary" variant="subhead">
                {t('Add wallet')}
              </Text>
            </Flex>
          </Button>
          <Button
            name={ElementName.Settings}
            testID={ElementName.Settings}
            onPress={onPressSettings}>
            <Flex row alignItems="center" gap="sm">
              <SettingsIcon color={theme.colors.textSecondary} height={24} width={24} />
              <Text color="textSecondary" variant="subhead">
                {t('Settings')}
              </Text>
            </Flex>
          </Button>
        </Flex>
      </Flex>
      <ActionSheetModal
        isVisible={showEditAccountModal}
        name={ModalName.Account}
        options={editAccountOptions}
        onClose={() => setShowEditAccountModal(false)}
      />
      {!!pendingRemoveAddress && (
        <RemoveAccountModal onCancel={onPressRemoveCancel} onConfirm={onPressRemoveConfirm} />
      )}
      <BottomSheetModal
        isVisible={showQRModal}
        name={ModalName.WalletQRCode}
        onClose={onCloseQrCode}>
        <WalletQRCode address={qrCodeAddress} />
      </BottomSheetModal>
    </Screen>
  )
}
