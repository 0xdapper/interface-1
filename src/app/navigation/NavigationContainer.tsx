import {
  createNavigationContainerRef,
  NavigationContainer as NativeNavigationContainer,
} from '@react-navigation/native'
import { AnyAction } from '@reduxjs/toolkit'
import React, { Dispatch, FC, useEffect, useState } from 'react'
import { Linking } from 'react-native'
import { useAppDispatch } from 'src/app/hooks'
import { DeepLink, openDeepLink } from 'src/features/deepLinking/handleDeepLink'
import { logScreenView } from 'src/features/telemetry'
import { Trace } from 'src/features/telemetry/Trace'

export const navigationRef = createNavigationContainerRef()

/** Wrapped `NavigationContainer` with telemetry tracing. */
export const NavigationContainer: FC = ({ children }) => {
  const [routeName, setRouteName] = useState<string | undefined>()
  const dispatch = useAppDispatch()

  useManageDeepLinks(dispatch)

  return (
    <NativeNavigationContainer
      ref={navigationRef}
      onReady={() => {
        setRouteName(navigationRef.getCurrentRoute()?.name)
      }}
      onStateChange={() => {
        const previousRouteName = routeName
        const currentRouteName = navigationRef.getCurrentRoute()?.name

        if (currentRouteName && previousRouteName !== currentRouteName) {
          logScreenView(currentRouteName)
        }

        setRouteName(currentRouteName)
      }}>
      <Trace screen={routeName}>{children}</Trace>
    </NativeNavigationContainer>
  )
}

export const useManageDeepLinks = (dispatch: Dispatch<AnyAction | AnyAction>) =>
  useEffect(() => {
    const handleDeepLink = (payload: DeepLink) => dispatch(openDeepLink(payload))
    const urlListener = Linking.addEventListener('url', (event: { url: string }) =>
      handleDeepLink({ url: event.url, coldStart: false })
    )
    const handleDeepLinkColdStart = async () => {
      const url = await Linking.getInitialURL()
      if (url) handleDeepLink({ url, coldStart: true })
    }

    handleDeepLinkColdStart()
    return urlListener.remove
  }, [dispatch])
