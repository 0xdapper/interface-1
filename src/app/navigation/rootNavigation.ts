import { navigationRef } from 'src/app/navigation/NavigationContainer'
import { RootParamList } from 'src/app/navigation/types'
import { logger } from 'src/utils/logger'

export async function navigate<RouteName extends keyof RootParamList>(
  ...args: undefined extends RootParamList[RouteName]
    ? [RouteName] | [RouteName, RootParamList[RouteName]]
    : [RouteName, RootParamList[RouteName]]
) {
  const [routeName, params] = args
  if (!navigationRef.isReady()) {
    logger.info('rootNavigation', 'navigate', 'Navigator was called before it was initialized')
    return
  }

  // Type assignment to `never` is a workaround until we figure out how to
  // type `createNavigationContainerRef` in a way that's compatible
  navigationRef.navigate(routeName as never, params as never)
}

export async function goBack() {
  if (!navigationRef.isReady()) {
    logger.info('rootNavigation', 'navigate', 'Navigator was called before it was initialized')
    return
  }

  navigationRef.goBack()
}
