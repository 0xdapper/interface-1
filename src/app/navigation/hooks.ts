import { NavigationContainerRefContext, NavigationContext } from '@react-navigation/core'
import { useCallback, useContext, useEffect } from 'react'
import { InteractionManager } from 'react-native'
import { useAppSelector } from 'src/app/hooks'
import { navigate as rootNavigate } from 'src/app/navigation/rootNavigation'
import { useAppStackNavigation, useExploreStackNavigation } from 'src/app/navigation/types'
import { baseCurrencyIds } from 'src/components/TokenSelector/hooks'
import {
  useExploreTokensTabLazyQuery,
  usePortfolioBalanceLazyQuery,
  usePortfolioBalancesLazyQuery,
  useTokenProjectsLazyQuery,
  useTopTokensLazyQuery,
  useTransactionListLazyQuery,
} from 'src/data/__generated__/types-and-hooks'
import { currencyIdToContractInput } from 'src/features/dataApi/utils'
import { getTokensOrderByValues } from 'src/features/explore/utils'
import { useActiveAccountAddress } from 'src/features/wallet/hooks'
import { selectTokensOrderBy } from 'src/features/wallet/selectors'
import { Screens } from 'src/screens/Screens'

/**
 * Utility hook to simplify navigating to Activity screen.
 * Preloads query needed to render transaction list.
 */
export function useEagerActivityNavigation() {
  const navigation = useAppStackNavigation()
  const [load] = useTransactionListLazyQuery()

  const preload = useCallback(
    (address: string) => {
      load({
        variables: {
          address,
        },
      })
    },
    [load]
  )

  const navigate = useCallback(() => navigation.navigate(Screens.Activity), [navigation])

  return { preload, navigate }
}

/**
 * Utility hook to simplify navigating to Activity screen.
 * Preloads query needed to render transaction list.
 */
export function useEagerExternalProfileNavigation() {
  const navigation = useExploreStackNavigation()

  const [load] = useTransactionListLazyQuery()

  const preload = useCallback(
    (address: string) => {
      load({ variables: { address } })
    },
    [load]
  )

  const navigate = useCallback(
    (address: string) => navigation.navigate(Screens.ExternalProfile, { address }),
    [navigation]
  )

  return { preload, navigate }
}

export function useEagerExternalProfileRootNavigation() {
  const [load] = useTransactionListLazyQuery()

  const preload = useCallback(
    (address: string) => {
      load({
        variables: {
          address,
        },
      })
    },
    [load]
  )

  const navigate = useCallback((address: string, callback?: () => void) => {
    rootNavigate(Screens.ExternalProfile, { address }).then(() => {
      callback?.()
    })
  }, [])

  return { preload, navigate }
}

/** Preloaded home screen queries that reload on active account change */
export function usePreloadedHomeScreenQueries() {
  const [loadPortfolioBalance] = usePortfolioBalanceLazyQuery()
  const [loadPortfolioBalances] = usePortfolioBalancesLazyQuery()
  const [loadTransactionHistory] = useTransactionListLazyQuery()

  const activeAccountAddress = useActiveAccountAddress()

  useEffect(() => {
    if (!activeAccountAddress) {
      return
    }

    loadPortfolioBalance({ variables: { owner: activeAccountAddress } })
    loadPortfolioBalances({ variables: { ownerAddress: activeAccountAddress } })
    loadTransactionHistory({ variables: { address: activeAccountAddress } })
  }, [activeAccountAddress, loadPortfolioBalance, loadPortfolioBalances, loadTransactionHistory])
}

/** Set of queries that should be preloaded, but can wait for idle time. */
export function useLowPriorityPreloadedQueries() {
  // Explore screen
  const [loadExploreTokens] = useExploreTokensTabLazyQuery()
  const orderBy = useAppSelector(selectTokensOrderBy)

  useEffect(() => {
    InteractionManager.runAfterInteractions(() => {
      const { serverOrderBy } = getTokensOrderByValues(orderBy)
      loadExploreTokens({ variables: { topTokensOrderBy: serverOrderBy } })
    })
    // only want to preload on app mount once
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [loadExploreTokens])

  // Token selector
  const [loadTopTokens] = useTopTokensLazyQuery()
  const [loadTokenProjects] = useTokenProjectsLazyQuery()

  useEffect(() => {
    const baseCurrencyContracts = baseCurrencyIds.map((id) => currencyIdToContractInput(id))

    InteractionManager.runAfterInteractions(() => {
      loadTokenProjects({ variables: { contracts: baseCurrencyContracts } })
      loadTopTokens()
    })
  }, [loadTopTokens, loadTokenProjects])
}

/**
 * Utility hook that checks if the caller is part of the navigation tree.
 *
 * Inspired by how the navigation library checks if the the navigation object exists.
 * https://github.com/react-navigation/react-navigation/blob/d7032ba8bb6ae24030a47f0724b61b561132fca6/packages/core/src/useNavigation.tsx#L18
 */
export function useIsPartOfNavigationTree() {
  const root = useContext(NavigationContainerRefContext)
  const navigation = useContext(NavigationContext)

  return navigation !== undefined || root !== undefined
}
