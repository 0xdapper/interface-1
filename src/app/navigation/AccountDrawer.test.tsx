import {
  DrawerDescriptorMap,
  DrawerNavigationHelpers,
} from '@react-navigation/drawer/lib/typescript/src/types'
import { DrawerNavigationState, ParamListBase } from '@react-navigation/routers'
import React from 'react'
import { PreloadedState } from 'redux'
import { AccountDrawer } from 'src/app/navigation/AccountDrawer'
import { RootState } from 'src/app/rootReducer'
import { initialModalState } from 'src/features/modals/modalSlice'
import { ModalName } from 'src/features/telemetry/constants'
import { mockWalletPreloadedState } from 'src/test/fixtures'
import { renderWithProviders } from 'src/test/render'

jest.mock('@react-navigation/drawer', () => ({
  useDrawerStatus: jest.fn().mockImplementation(() => 'open'),
}))

const preloadedState = {
  ...mockWalletPreloadedState,
  modals: {
    ...initialModalState,
    [ModalName.AccountSwitcher]: { isOpen: true },
  },
} as unknown as PreloadedState<RootState>

describe(AccountDrawer, () => {
  it('renders correctly', () => {
    const tree = renderWithProviders(
      <AccountDrawer
        descriptors={{} as DrawerDescriptorMap}
        navigation={{} as DrawerNavigationHelpers}
        state={{} as DrawerNavigationState<ParamListBase>}
      />,
      {
        preloadedState,
      }
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
