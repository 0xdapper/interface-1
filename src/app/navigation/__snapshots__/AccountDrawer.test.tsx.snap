// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDrawer renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "display": "flex",
        "flex": 1,
      },
      Object {
        "paddingBottom": undefined,
        "paddingTop": undefined,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "paddingHorizontal": 24,
          "paddingTop": 24,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 999999,
              "borderWidth": 0,
              "display": "flex",
              "position": "relative",
            },
            Object {
              "padding": 0,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "height": 40,
                "width": 40,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              clip={true}
              mask={
                <UniconMask
                  attributeData={
                    Object {
                      "0": "#5CFE9D",
                      "1": "#B8C3B7",
                      "2": Array [
                        Object {
                          "path": "M34 20C34 28.8366 26.8366 36 18 36C9.16344 36 2 28.8366 2 20L2 -3.8147e-06L34 -1.01717e-06V20Z",
                        },
                      ],
                      "3": Array [
                        Object {
                          "fillType": "evenOdd",
                          "path": "M16 8C12.2937 6.62854 9.37146 3.70632 8 0C6.62854 3.70632 3.70632 6.62854 0 8C3.70632 9.37146 6.62854 12.2937 8 16C9.37146 12.2937 12.2937 9.37146 16 8ZM8 11C9.65685 11 11 9.65685 11 8C11 6.34315 9.65685 5 8 5C6.34315 5 5 6.34315 5 8C5 9.65685 6.34315 11 8 11Z",
                        },
                      ],
                    }
                  }
                  size={40}
                />
              }
            >
              <View
                transform={
                  Array [
                    Object {
                      "scale": 1.0810810810810811,
                    },
                  ]
                }
              >
                <View
                  height={37}
                  width={37}
                  x={0}
                  y={0}
                >
                  <View
                    colors={
                      Array [
                        "#5CFE9D",
                        "#B8C3B7",
                      ]
                    }
                    end={null}
                    start={null}
                  />
                </View>
                <View
                  color="#EDE590"
                  cx={18.5}
                  cy={-13.36111111111111}
                  r={30.833333333333336}
                >
                  <View
                    blur={15}
                  />
                </View>
              </View>
            </View>
            <View
              clip={true}
              mask={
                <UniconMask
                  attributeData={
                    Object {
                      "0": "#5CFE9D",
                      "1": "#B8C3B7",
                      "2": Array [
                        Object {
                          "path": "M34 20C34 28.8366 26.8366 36 18 36C9.16344 36 2 28.8366 2 20L2 -3.8147e-06L34 -1.01717e-06V20Z",
                        },
                      ],
                      "3": Array [
                        Object {
                          "fillType": "evenOdd",
                          "path": "M16 8C12.2937 6.62854 9.37146 3.70632 8 0C6.62854 3.70632 3.70632 6.62854 0 8C3.70632 9.37146 6.62854 12.2937 8 16C9.37146 12.2937 12.2937 9.37146 16 8ZM8 11C9.65685 11 11 9.65685 11 8C11 6.34315 9.65685 5 8 5C6.34315 5 5 6.34315 5 8C5 9.65685 6.34315 11 8 11Z",
                        },
                      ],
                    }
                  }
                  overlay={true}
                  size={40}
                />
              }
            >
              <View
                color="#000000"
                height={37}
                opacity={0.08}
                width={37}
                x={0}
                y={0}
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "width": 12,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexShrink": 1,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            allowFontScaling={true}
            ellipsizeMode="tail"
            maxFontSizeMultiplier={1.4}
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#0D111C",
                  "display": "flex",
                  "fontFamily": "Inter-Medium",
                  "fontSize": 20,
                  "lineHeight": 24,
                },
              ]
            }
            testID="address-display/name/Test Account"
          >
            Test Account
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 5,
              "left": 5,
              "right": 5,
              "top": 5,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "display": "flex",
              "opacity": 1,
            }
          }
          testID="copy"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              allowFontScaling={true}
              maxFontSizeMultiplier={1.4}
              style={
                Array [
                  Object {
                    "color": "#505A78",
                    "display": "flex",
                    "fontFamily": "Inter-Medium",
                    "fontSize": 14,
                    "lineHeight": 20,
                  },
                ]
              }
            >
              0xâ€‹f39F...2266
            </Text>
            <View
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "height": 4,
                    "width": 4,
                  },
                ]
              }
            />
            <svg
              color="#505A78"
              data-file-name="SvgCopySheets"
              height={14}
              width={14}
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#D2D9EE",
          "borderBottomWidth": 0.25,
          "display": "flex",
          "height": 1,
          "marginVertical": 16,
          "overflow": "visible",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "paddingBottom": 24,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 5,
          "left": 5,
          "right": 5,
          "top": 5,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "display": "flex",
          "opacity": 1,
        }
      }
      testID="manage-connections"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "width": 40,
              },
            ]
          }
        >
          <svg
            color="#505A78"
            data-file-name="SvgGlobal"
            height={24}
            width={24}
          />
        </View>
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "flexGrow": 0,
                "flexShrink": 0,
                "height": 8,
                "width": 8,
              },
            ]
          }
        />
        <Text
          allowFontScaling={true}
          maxFontSizeMultiplier={1.4}
          style={
            Array [
              Object {
                "color": "#0D111C",
                "display": "flex",
                "fontFamily": "Inter-Medium",
                "fontSize": 16,
                "lineHeight": 24,
              },
            ]
          }
        >
          Manage connections
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexGrow": 0,
            "flexShrink": 0,
            "height": 24,
          },
        ]
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 5,
          "left": 5,
          "right": 5,
          "top": 5,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "display": "flex",
          "opacity": 1,
        }
      }
      testID="settings"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "width": 40,
              },
            ]
          }
        >
          <svg
            color="#505A78"
            data-file-name="SvgSettings"
            height={24}
            width={24}
          />
        </View>
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "flexGrow": 0,
                "flexShrink": 0,
                "height": 8,
                "width": 8,
              },
            ]
          }
        />
        <Text
          allowFontScaling={true}
          maxFontSizeMultiplier={1.4}
          style={
            Array [
              Object {
                "color": "#0D111C",
                "display": "flex",
                "fontFamily": "Inter-Medium",
                "fontSize": 16,
                "lineHeight": 24,
              },
            ]
          }
        >
          Settings
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexGrow": 0,
            "flexShrink": 0,
            "height": 24,
          },
        ]
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 5,
          "left": 5,
          "right": 5,
          "top": 5,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "display": "flex",
          "opacity": 1,
        }
      }
      testID="get-help"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "width": 40,
              },
            ]
          }
        >
          <svg
            color="#505A78"
            data-file-name="SvgHelp"
            height={24}
            width={24}
          />
        </View>
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "flexGrow": 0,
                "flexShrink": 0,
                "height": 8,
                "width": 8,
              },
            ]
          }
        />
        <Text
          allowFontScaling={true}
          maxFontSizeMultiplier={1.4}
          style={
            Array [
              Object {
                "color": "#0D111C",
                "display": "flex",
                "fontFamily": "Inter-Medium",
                "fontSize": 16,
                "lineHeight": 24,
              },
            ]
          }
        >
          Get help
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexGrow": 1,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#D2D9EE",
          "borderBottomWidth": 0.25,
          "display": "flex",
          "height": 1,
          "marginBottom": 16,
          "overflow": "visible",
        },
      ]
    }
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "display": "flex",
        "marginBottom": 0,
        "marginLeft": 24,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "#D2D9EE",
              "borderRadius": 999999,
              "borderWidth": 1,
              "display": "flex",
              "justifyContent": "center",
              "padding": 8,
            },
          ]
        }
      >
        <svg
          color="#505A78"
          data-file-name="SvgPlus"
          height={12}
          width={12}
        />
      </View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
      />
      <Text
        allowFontScaling={true}
        maxFontSizeMultiplier={1.4}
        style={
          Array [
            Object {
              "color": "#0D111C",
              "display": "flex",
              "fontFamily": "Inter-Medium",
              "fontSize": 16,
              "lineHeight": 24,
            },
          ]
        }
      >
        Add another wallet
      </Text>
    </View>
  </View>
</View>
`;
